src/LineageTree/lineageTree.py

```diff
@@ -2400,6 +2403,445 @@ def get_cells_at_t_from_root(self, r: [int, list], t: int = None) -> list:
+            )
+            pos_cycle1 = np.transpose(np.dot(R, pos_cycle1.T) + t)
+
+        N = len(nodes1_cycle)
+        M = len(nodes2_cycle)
+
+        dist_mat = distance.cdist(pos_cycle1, pos_cycle2, 'euclidean')
+
+        if fast:
+            if centered_band:
+                slope, intercept = self.__calculate_diag_line(dist_mat)
+                path, cost_mat, final_cost = self.__dp(dist_mat, start_d, back_d, w, slope, intercept, use_absolute=False)
+            else:
+                path, cost_mat, final_cost = self.__dp(dist_mat, start_d, back_d, w, use_absolute=True)
+        else:
+            path, cost_mat, final_cost = self.__dp(dist_mat, start_d, back_d)
+
+        cost = final_cost / len(path)
+
+        if cost_mat_p:
+            return cost, path, cost_mat, pos_cycle1, pos_cycle2
+        else:
+            return cost, path
+
+    def plot_dtw_heatmap(
+        self,
+        nodes1: int,
+        nodes2: int,
+        threshold: int=1000,
+        regist: bool=True,
+        start_d: int=0,
+        back_d: int=0,
+        fast: bool=False,
+        w: int=0,
+        centered_band: bool=True,
+    ) -> (float, plt.figure):
+        """
+        Plot DTW cost matrix between two cell cycles in heatmap format
+
+            Args:
+                nodes1 (int): node to compare distance
+                nodes2 (int): node to compare distance
+                start_d (int): start delay
+                back_d (int): end delay
+                fast (boolean): True if the user wants to run the fast algorithm with window restrains
+                w (int): window size
+                centered_band (boolean): if running the fast algorithm, True if the windown is centered
+
+            Returns:
+                (float) DTW distance
+                (figure) Heatmap of cost matrix with opitimal path
+        """
+        cost, path, cost_mat, pos_cycle1, pos_cycle2 = self.calculate_dtw(
+            nodes1,
+            nodes2,
+            threshold,
+            regist,
+            start_d,
+            back_d,
+            fast,
+            w,
+            centered_band,
+            cost_mat_p=True,
+        )
+
+        fig = plt.figure(figsize=(8, 6))
+        ax = fig.add_subplot(1, 1, 1)
+        im = ax.imshow(
+            cost_mat, cmap="viridis", origin="lower", interpolation="nearest"
+        )
+        plt.colorbar(im)
+        ax.set_title("Heatmap of DTW Cost Matrix")
+        ax.set_xlabel("Tree 1")
+        ax.set_ylabel("tree 2")
+        x_path, y_path = zip(*path)
+        ax.plot(y_path, x_path, color="black")
+
+        return cost, fig
+
+    def plot_dtw_trajectory(
+        self,
+        nodes1: int,
+        nodes2: int,
+        threshold: int=1000,
+        regist: bool=True,
+        start_d: int=0,
+        back_d: int=0,
+        fast: bool=False,
+        w: int=0,
+        centered_band: bool=True,
+        three_dim: bool=True,
+        alig: bool=False,
+    ) -> (float, plt.figure):
```

Comment:
What if we deleted the three_dim variable and only kept the projection where the user could write:
'3d' : for the 3d visualization
'xy' or None (default) : 2D projection of axis x and y
'xz' : 2D projection of axis x and z
'yz' : 2D projection of axis y and z
'pca' : PCA projection
Because the way that it's now three_dim default is True, so if the user only types plot_dtw_trajectory(nodes1, nodes2, projection = 'xy') he's still gonna get the 3D plot, so I think it can be confusing.

Reply:
I think you're right. Do it that way then